core/src/test/java/org/elasticsearch/client/transport/TransportClientIT.java

```diff
@@ -97,4 +106,47 @@ public void testThatTransportClientSettingCannotBeChanged() {
             assertThat(Client.CLIENT_TYPE_SETTING_S.get(settings), is("transport"));
         }
     }
+
+    /**
+     * test that when plugins are provided that want to register
+     * {@link NamedWriteable}, those are also made known to the
+     * {@link NamedWriteableRegistry} of the transport client
+     */
+    public void testPluginNamedWriteablesRegistered() {
+        Settings baseSettings = Settings.builder()
+                .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir())
+                .build();
+        try (TransportClient client = new MockTransportClient(baseSettings, pluginList(MockPlugin.class))) {
+            NamedWriteableRegistry registry = client.injector.getInstance(NamedWriteableRegistry.class);
```

Comment:
I'm not sure we should be adding more references to the injector. Either we should make the `NamedWriteableRegistry` a package private member of the client or we should just serialize the thing somehow.